ls: don't use colors if stdout is not a TTY

Here is the relevant snippet in `configure`, at least on my Mac running Catalina

```
if (
   am_has_slept=no
   for am_try in 1 2; do
     echo "timestamp, slept: $am_has_slept" > conftest.file
     set X `ls -Lt "$srcdir/configure" conftest.file 2> /dev/null`
     if test "$*" = "X"; then
	# -L didn't work.
	set X `ls -t "$srcdir/configure" conftest.file`
     fi
     if test "$*" != "X $srcdir/configure conftest.file" \
	&& test "$*" != "X conftest.file $srcdir/configure"; then

	# If neither matched, then we have a broken ls.  This can happen
	# if, for instance, CONFIG_SHELL is bash and it inherits a
	# broken ls alias from the environment.  This has actually
	# happened.  Such a system could not be considered "sane".
	as_fn_error $? "ls -t appears to fail.  Make sure there is not a broken
  alias in your environment" "$LINENO" 5
     fi
     if test "$2" = conftest.file || test $am_try -eq 2; then
       break
     fi
     # Just in case.
     sleep 1
     am_has_slept=yes
   done
   test "$2" = conftest.file
   )
then
   # Ok.
```

I get the following error message when I attempt to compile this using coreutils `ls`

```
checking for a BSD-compatible install... /Users/kevin/local/coreutils/bin/install -c
checking whether build environment is sane... configure: error: ls -t appears to fail.  Make sure there is not a broken
  alias in your environment
configure: error: newly created file is older than distributed files!
Check your system clock
```

I pulled out most of it, but here is the full thing.

```
checking whether to enable maintainer-specific portions of Makefiles... yes
checking build system type... error: Found argument '-p' which wasn't expected, or isn't valid in this context

USAGE:
    uname [OPTION]...

For more information try --help
powerpc-apple-darwin19.6.0
checking host system type... x86_64-apple-darwin19.6.0
checking for a BSD-compatible install... /Users/kevin/local/coreutils/bin/install -c
checking whether build environment is sane... configure: error: ls -t appears to fail.  Make sure there is not a broken
  alias in your environment
configure: error: newly created file is older than distributed files!
Check your system clock
```

--- a/src/uu/ls/src/ls.rs
+++ b/src/uu/ls/src/ls.rs
@@ -619,7 +619,7 @@ fn display_file_name(
     let mut width = UnicodeWidthStr::width(&*name);
 
     let color = match options.opt_str("color") {
-        None => stdout_isatty(),
+        None => atty::is(atty::Stream::Stdout),
         Some(val) => match val.as_ref() {
             "always" | "yes" | "force" => true,
             "auto" | "tty" | "if-tty" => stdout_isatty(),

--- a/src/uu/ls/src/ls.rs
+++ b/src/uu/ls/src/ls.rs
@@ -619,7 +619,7 @@ fn display_file_name(
     let mut width = UnicodeWidthStr::width(&*name);
 
     let color = match options.opt_str("color") {
-        None => true,
+        None => stdout_isatty(),
         Some(val) => match val.as_ref() {
             "always" | "yes" | "force" => true,
             "auto" | "tty" | "if-tty" => stdout_isatty(),

